### ISSUE TITLE
Problem concerning tip order

### ISSUE THREAD

**Issue by MichelleKendall** [thibautjombart/exploratree#8]
I've just been looking through the code closely and unfortunately there's a problem with simply calling "CK.metric" for each tree. That's because in order to compare the vectors properly we need to traverse the tips in the same order for each tree. This is done correctly inside multi.dist using the "labelmatch" permutation, but the single tree function is not equipped for it. I don't know whether it will be more efficient to apply the permutation within the function (as I've done in multi.dist) or to the vector after it has been calculated, but I'm afraid it will need doing at some stage in the process. Let me know if you want me to implement this, or whether it's easier to just use multi.dist now that it (hopefully) doesn't contain any stray brackets.

Cheers,

Michelle

**Comment by thibautjombart** [comment]
Hi Michelle, 
if there is a way to order the outcome vector consistently across trees, it would be great to have it embedded in CK.metric. The first thing users will try is compare outputs of CK.metric on trees describing the same taxa, so the ordering problem will be a recurrent one. Does it make sense?

**Comment by MichelleKendall** [comment]
Hi, yes that makes sense and I think I can see how to do it. If we can rely on the function "order()" to always behave the same under the same inputs (?) then we just need to include
permuation <- order(tree$tip.label)
and do a bit of rearrangement at the end. I've actually already written code that will do that, let me see if I can find it. Can you see any logical problem with using the order() function, for example if the tip labels are unusual?

**Comment by MichelleKendall** [comment]
OK, I'm pretty sure I've got that working. I've uploaded it as a separate file, "tiporder correction", because I'm not sure how the version control works on here and I don't want to break anything! But it should be fine to replace the existing code with it. I can't think of any scenario where the ordering function would behave badly apart from if two tips have the same label, but that would cause problems in any version I think. Perhaps we could include a warning message
